src/thor/costmatrix.cc

```diff
@@ -78,40 +80,49 @@ float CostMatrix::GetCostThreshold(const float max_matrix_distance) {
 // Clear the temporary information generated during time + distance matrix
 // construction.
 void CostMatrix::clear() {
-  // Clear the target edge markings
-  for (auto& iter : *targets_) {
-    iter.second.clear();
-    iter.second.resize(0);
-    iter.second.shrink_to_fit();
+  // resize all relevant structures down to 20 locations
+  if (source_count_ > kMaxLocationCache) {
+    source_edgelabel_.resize(kMaxLocationCache);
+    source_edgelabel_.shrink_to_fit();
+    source_adjacency_.resize(kMaxLocationCache);
+    source_adjacency_.shrink_to_fit();
+    source_edgestatus_.resize(kMaxLocationCache);
+    source_edgestatus_.shrink_to_fit();
+  }
+  if (target_count_ > kMaxLocationCache) {
+    target_edgelabel_.resize(kMaxLocationCache);
+    target_edgelabel_.shrink_to_fit();
+    target_adjacency_.resize(kMaxLocationCache);
+    target_adjacency_.shrink_to_fit();
+    target_edgestatus_.resize(kMaxLocationCache);
+    target_edgestatus_.shrink_to_fit();
   }
-  targets_->clear();
 
-  // Clear all source adjacency lists, edge labels, and edge status
-  // Resize and shrink_to_fit so all capacity is reduced.
-  source_adjacency_.clear();
-  source_adjacency_.resize(0);
-  source_adjacency_.shrink_to_fit();
-  target_adjacency_.clear();
-  target_adjacency_.resize(0);
-  target_adjacency_.shrink_to_fit();
-  source_edgelabel_.clear();
-  source_edgelabel_.resize(0);
-  source_edgelabel_.shrink_to_fit();
-  target_edgelabel_.clear();
-  target_edgelabel_.resize(0);
-  target_edgelabel_.shrink_to_fit();
-  source_edgestatus_.clear();
-  source_edgestatus_.resize(0);
-  source_edgestatus_.shrink_to_fit();
-  target_edgestatus_.clear();
-  target_edgestatus_.resize(0);
-  target_edgestatus_.shrink_to_fit();
+  auto reservation = clear_reserved_memory_ ? 0U : max_reserved_labels_count_;
```

Comment:
and for those 20 locations, we keep a default of 2 Mio around per location. for labels that should be around 20MB per location.

Reply:
2 Mio is quite a lot.. but also it's dijkstra.. well, it's configurable for the server